### PR TITLE
cpu/stm32: Implement periph/gpio_ll{,_irq} except for STM32F1

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Implements `gpio_ll` and `gpio_ll_irq` for STM32* except for STM32F1.

### Testing procedure

```
make BOARD=nucleo-<foo> -C tests/periph_gpio_ll
```

#### Testing State

- [x] STM32F0 (bluepill-stm32f030c8, @maribu)
- [ ] ~~STM32F1 (Cannot test, no implementation yet)~~
- [ ] STM32F2
- [x] STM32F3 (Nucleo-F303RE, @maribu)
- [x] STM32F4 (Nucleo-F446RE, @maribu)
- [x] STM32F7 (Nucleo-F767ZI, @maribu)
- [ ] STM32G0
- [x] STM32G4
- [x] STM32L0 (Nucleo-L011K4, @maribu)
- [ ] STM32L1
- [x] STM32L4 (Nucleo-L432KC, @maribu)
- [x] STM32L5 (Nucleo-L552ze-q, @krzysztof-cabaj)
- [ ] STM32MP1
- [ ] STM32U5
- [ ] STM32WB
- [ ] STM32WL

### Issues/PRs references

Depends on and includes: https://github.com/RIOT-OS/RIOT/pull/16787

### CODE REVIEW

**Comment by krzysztof-cabaj** [comment]
Hi. I tested this PR using Nucleo-L552ze-q. Test output:

```
<snip>
TEST SUCCEEDED
{ "threads": [{ "name": "main", "stack_size": 1536, "stack_used": 440 }]}
```

**Comment by bergzand** [comment]
Tested the nucleo-g474re:
```
2022-05-17 10:21:58,552 # TEST SUCCEEDED
2022-05-17 10:21:58,558 # { "threads": [{ "name": "main", "stack_size": 1536, "stack_used": 432 }]}
```

File: cpu/stm32/periph/gpio_ll_irq.c [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    isr_ctx[pin].arg = arg;
+
+    /* enable clock of the SYSCFG module for EXTI configuration */
+#if !defined(CPU_FAM_STM32WB) && !defined(CPU_FAM_STM32MP1) && \
+    !defined(CPU_FAM_STM32WL)
+#ifdef CPU_FAM_STM32F0
+    periph_clk_en(APB2, RCC_APB2ENR_SYSCFGCOMPEN);
+#elif defined(CPU_FAM_STM32G0)
+    periph_clk_en(APB12, RCC_APBENR2_SYSCFGEN);
+#elif defined(CPU_FAM_STM32U5)
```
Comment (benpicco) [inline cpu/stm32/periph/gpio_ll_irq.c:204]:
```suggestion
#elif defined(RCC_APB3ENR_SYSCFGEN)
```

**Review by benpicco — APPROVED** [review]
I wonder if we could make porting a bit easier though while we're at it

**Comment by maribu** [comment]
Thanks :)

**Pull request merged**